src/typeahead/typeahead-window.ts

```diff
@@ -39,6 +41,12 @@ export class NgbTypeaheadWindow implements OnInit {
   activeIdx = 0;
 
   /**
+   *  An optional id for the typeahead. The id should be unique.
+   *  If not provided, it will be auto-generated.
+   */
+  @HostBinding('attr.id') @Input() id: string = undefined;
```

Comment:
Also the comment seems to be invalid as we are not auto-generating id when one is not provided.